docs(c8run): mention no-db support for C8Run (#6686)
camunda/camunda-docs: docs/self-managed/concepts/no-secondary-storage.md

## When to use this mode

Use `noSecondaryStorage` mode **only** in specific situations, such as:

> - **Local development with C8Run**: Run Zeebe locally without secondary storage components by setting the `camunda.database.type=none` environment variable.

Reviewer: CC @zaycodes Note that we do not refer to it as C8 Run in the docs -- fully Camunda 8 Run. I realize this isn't in the glossary so it is easy to miss -- I'll add it 😄

> Consider `noSecondaryStorage` mode **only** in specific situations, such as:
> 
> - **Local development with C8Run**: For developers who want to run Zeebe locally without the overhead of secondary storage components, C8Run supports this mode by setting the `camunda.database.type=none` environment variable.
> - **Specialized technical requirements** that prevent the use of secondary storage.
> - **Temporary migration scenarios** where minimal orchestration functionality is needed during a transition period.
> - **Resource-constrained environments** where deploying the full platform is not feasible.

Reviewer 2: ```suggestion
Use `noSecondaryStorage` mode **only** in specific cases, such as:

- **Local development with C8Run**: Run Zeebe locally without secondary storage components by setting the `camunda.database.type=none` environment variable.
- **Specialized technical requirements** that prevent the use of secondary storage.
- **Temporary migration scenarios** where minimal orchestration functionality is needed during a transition period.
- **Resource-constrained environments** where deploying the full platform is not feasible.
```
